pcl_conversions/include/pcl_conversions/pcl_conversions.h

```diff
@@ -717,191 +724,193 @@ namespace pcl {
-        stream.next(pfs);
-        pcl_conversions::toPCL(pfs, m.fields);
-        stream.next(m.is_bigendian);
-        stream.next(m.point_step);
-        stream.next(m.row_step);
-        stream.next(m.data);
-        stream.next(m.is_dense);
-      }
-
-      inline static uint32_t serializedLength(const pcl::PCLPointCloud2& m)
-      {
-        uint32_t length = 0;
-
-        std_msgs::Header header;
-        pcl_conversions::fromPCL(m.header, header);
-        length += serializationLength(header);
-        length += 4; // height
-        length += 4; // width
-        std::vector<sensor_msgs::PointField> pfs;
-        pcl_conversions::fromPCL(m.fields, pfs);
-        length += serializationLength(pfs); // fields
-        length += 1; // is_bigendian
-        length += 4; // point_step
-        length += 4; // row_step
-        length += 4; // data's size
-        length += m.data.size() * sizeof(pcl::uint8_t);
-        length += 1; // is_dense
-
-        return length;
-      }
-    };
-
-    /*
-     * Provide a custom serialization for pcl::PCLPointField
-     */
-    template<>
-    struct Serializer<pcl::PCLPointField>
-    {
-      template<typename Stream>
-      inline static void write(Stream& stream, const pcl::PCLPointField& m)
-      {
-        stream.next(m.name);
-        stream.next(m.offset);
-        stream.next(m.datatype);
-        stream.next(m.count);
-      }
-
-      template<typename Stream>
-      inline static void read(Stream& stream, pcl::PCLPointField& m)
-      {
-        stream.next(m.name);
-        stream.next(m.offset);
-        stream.next(m.datatype);
-        stream.next(m.count);
-      }
-
-      inline static uint32_t serializedLength(const pcl::PCLPointField& m)
-      {
-        uint32_t length = 0;
-
-        length += serializationLength(m.name);
-        length += serializationLength(m.offset);
-        length += serializationLength(m.datatype);
-        length += serializationLength(m.count);
-
-        return length;
-      }
-    };
-
-    /*
-     * Provide a custom serialization for pcl::PCLHeader
-     */
-    template<>
-    struct Serializer<pcl::PCLHeader>
-    {
-      template<typename Stream>
-      inline static void write(Stream& stream, const pcl::PCLHeader& m)
-      {
-        std_msgs::Header header;
-        pcl_conversions::fromPCL(m, header);
-        stream.next(header);
-      }
-
-      template<typename Stream>
-      inline static void read(Stream& stream, pcl::PCLHeader& m)
-      {
-        std_msgs::Header header;
-        stream.next(header);
-        pcl_conversions::toPCL(header, m);
-      }
-
-      inline static uint32_t serializedLength(const pcl::PCLHeader& m)
-      {
-        uint32_t length = 0;
-
-        std_msgs::Header header;
-        pcl_conversions::fromPCL(m, header);
-        length += serializationLength(header);
-
-        return length;
-      }
-    };
-  } // namespace ros::serialization
-
-} // namespace ros
+//     template<>
```

Comment:
This is for type masquerading?

Reply:
Hm good question, I don't have an answer to that.

Reply:
@clalancette that's what this is for, correct?